routing/restriction_loader.cpp

```diff
@@ -1,9 +1,45 @@
 #include "routing/restriction_loader.hpp"
-#include "routing/routing_serialization.hpp"
+#include "routing/restrictions_serialization.hpp"
+#include "routing/road_index.hpp"
+
+#include "base/stl_helpers.hpp"
+
+namespace
+{
+using namespace routing;
+
+/// \returns if features |r1| and |r2| have a common end returns its joint id.
+/// If not, returnfs Joint::kInvalidId.
+/// \note It's possible that the both ends of |r1| and |r2| has common joint ids.
+/// In that case returns any of them.
+/// \note In general case ends of features don't have to be joints. For example all
+/// loose feature ends aren't joints. But if ends of r1 and r2 are connected at this
```

Comment:
I'm taking here about features with  loose ends. If so, yes. We have a lot features two-way road features with loose ends:
http://www.openstreetmap.org/#map=19/55.44064/37.77966

But more then that. I've found one one-way (deadlock) feature with loose end in Moscow! @Zverik fixed it as soon as I found it.